sqllogictest-engines/src/external.rs

```diff
@@ -98,7 +98,11 @@ impl AsyncDB for ExternalDriver {
             sql: sql.to_string(),
         };
         let input = serde_json::to_string(&input)?;
-        self.stdin.write_all(input.as_bytes()).await?;
+        match &mut self.stdin {
+            Some(stdin) => stdin.write_all(input.as_bytes()).await?,
+            None => return Err(io::Error::from(io::ErrorKind::UnexpectedEof).into()),
+        };
```

Comment:
Shall we directly panic if it's already shut down, just like

https://github.com/risinglightdb/sqllogictest-rs/blob/3a603d79ffb048ea971fc2c7f3017f7518f90890/sqllogictest-engines/src/postgres.rs#L50